addons/mail/tests/test_res_partner.py

```diff
@@ -544,3 +544,23 @@ def test_partner_merge_wizards(self):
         all_msg = p2_msg_ids_init + p1_msg_ids_init + p1_msg1
         self.assertEqual(len(p2.message_ids), len(all_msg) + 1, 'Should have original messages + a log')
         self.assertTrue(all(msg in p2.message_ids for msg in all_msg))
+
+    def test_subtype_parent_same_model(self):
+        subtype_1 = self.env['mail.message.subtype'].create({
+            'name': '',
+            'res_model': 'res.partner',
+            'default': True,
+        })
+        self.env['mail.message.subtype'].create({
+            'name': '',
+            'res_model': 'res.partner',
+            'default': True,
+            'parent_id': subtype_1.id,
+            'relation_field': 'parent_id',
+        })
+
+        Partner = self.env['res.partner']
+
+        p1 = Partner.with_user(self.user_employee).create({'name': 'Customer1', 'email': 'test1@test.example.com'})
+        p2 = Partner.with_user(self.user_admin).create({'name': 'Customer1 Child', 'email': 'test1_child@test.example.com', 'parent_id': p1.id})
+        self.assertIn(self.partner_employee, p2.message_follower_ids.partner_id, 'The partner from the parent has not been added as follower.')
```

Comment:
Would put this next to `test_auto_subscribe_defaults`, or if doable even rework that method directly to test with a "recursive" container model, using a loop over test models or something.
And then add a test for the "recursive" part.